api/src/main/java/org/apache/cloudstack/api/command/admin/vm/MigrateVMCmd.java

```diff
@@ -155,6 +161,8 @@ public void execute() {
                 throw new InvalidParameterValueException("The specified host(" + destinationHost.getName() + ") is not suitable to migrate the VM, please specify another one");
             }
             CallContext.current().setEventDetails("VM Id: " + getVirtualMachineId() + " to host Id: " + getHostId());
+        } else if (! isAutoSelect()) {
+            throw new InvalidParameterValueException("Please specify a host or storage as destination, or pass 'autoselect=true' to automatically select a destination host which do not require storage migration");
```

Comment:
@weizhouapache just to check, auto select here picks the suitable host for VM migration. In the similar way, any possibility to pick suitable storage ? If so, can this API have both parameters: autoselecthost & autoselectstorage (only one of them should be true).

Reply:
@sureshanaparti
this pr works only if storage migraiotn is not required.
it can be improved in a new pr.